lib/browser/storage.js

```diff
@@ -5,10 +5,18 @@ try {
   hasStorage = "localStorage" in window;
   // Attempt to access localStorage
   localStorage.length;
+
+  // Attempt to store and read entries from the local storage to detect Private
+  // Mode on Safari on iOS (see #49)
+  var test = 'tusSupport';
+  localStorage.setItem(test, test);
+  localStorage.removeItem(test);
```

Comment:
When test doesn't exist in the first place you'd attempt to set and item with the value `undefined`, then an error would be thrown even when there was local storage support